File: sic-api/Model/Storage/StorageUploadReference.cs
using Microsoft.EntityFrameworkCore;

namespace sic_api.Model.Storage;

public sealed class StorageUploadReference
{
    public EntityState State { get; set; } = EntityState.Detached;
    public Guid Id { get; set; }
    public Guid? UploadGroupId { get; set; }
    public bool IsStreaming { get; set; }
    public bool IsActive { get; set; }
    public string FileName { get; set; } = default!;
    public string ContentType { get; set; } = default!;
    public long FileSize { get; set; }
    public string Visibility { get; set; } = default!;
    public string AccessUrl { get; set; } = default!;
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/ex/StorageUploadReference.java
package com.softinter.sicapi.entity.ex;

import java.util.UUID;

import com.softinter.sicapi.entity.enums.EntityState;

import lombok.Data;

@Data
public class StorageUploadReference {
    private UUID id;
    private UUID uploadGroupId;
    private String fileName;
    private String contentType;
    private Long fileSize;
    private String accessUrl;
    private Boolean isActive;
    private Boolean isStreaming = false;   
    private String visibility;             
    private Integer state;
}
